lib/chess/core/pieces/base_piece.rb

```diff
@@ -58,10 +58,19 @@ def roll_back_cell
           @current_cell = back_cell
         end
 
+        def back_to_board
+          @captured = false
+          @current_cell.occupy_with(self)
+        end
+
         def captured?
           @captured
         end
 
+        def takken
+          @captured = true
+        end
```

Comment:
IMO I'd expect that a method called `taken` returns a boolean if the piece was taken or not. If the method is meant to mark the piece as captured, I suggest you call it with a verb, like `become_captured`
```suggestion
        def become_captured
          @captured = true
        end
```

Reply:
yeah, done!